### PR TITLE
WIP: VTOL_TAKEOFF, VTOL_LAND commands and VT_FORCE_VTOL param

### PR DESCRIPTION
**Pull request by sanderux**
Re-implement support for MAV_CMD_NAV_VTOL_TAKEOFF and MAV_CMD_NAV_VTOL_LAND
Replaces https://github.com/PX4/Firmware/pull/3699

Fixes https://github.com/PX4/Firmware/issues/3492
Fixes https://github.com/PX4/Firmware/issues/3712

### CODE REVIEW

**Comment by sanderux** [comment]
Related QGC update: https://github.com/mavlink/qgroundcontrol/pull/2769

**Comment by sanderux** [comment]
@AndreasAntener can you review this?
I will flight test saturday

**Comment by sanderux** [comment]
added VT_FORCE_VTOL param for airframes that are incapable of FW landings

**Comment by sanderux** [comment]
@AndreasAntener i will test fly this tomorrow, i think it's ready for merge (for completeness i will limit transitions commands to only when applicable (only issue MC trans when !is_rotary_wing and vice versa)

File: src/modules/navigator/mission.cpp [hunk 1]
```diff
@@ -629,6 +652,18 @@ Mission::do_need_takeoff()
 bool
 Mission::do_need_move_to_land()
 {
+	if(_mission_item.nav_cmd == NAV_CMD_VTOL_LAND || (_mission_item.nav_cmd == NAV_CMD_LAND && _param_force_vtol.get())){
```
Comment (AndreasAntener) [inline src/modules/navigator/mission.cpp]: This adds a side effect to a check method. I would like to give this it's own work item that would be handled before need to move to land check. So adding it before this line: https://github.com/PX4/Firmware/pull/3720/files#diff-9b5a77642a641b107c8f8a966de34acdR466

File: src/modules/navigator/mission.cpp [hunk 2]
```diff
@@ -440,7 +444,21 @@ Mission::set_mission_items()
 		}
 
 		/* if we just did a takeoff navigate to the actual waypoint now */
-		if (_work_item_type == WORK_ITEM_TYPE_TAKEOFF) {
+		if (_work_item_type == WORK_ITEM_TYPE_TAKEOFF) {			
+
+			/* handle VTOL TAKEOFF command */
+			if(_takeoff_vtol_transition){
```
Comment (AndreasAntener) [inline src/modules/navigator/mission.cpp]: I have an interesting question here. Do you want to navigate to the takeoff waypoint before or after the transition? ;)
And another one: is it really the idea of this waypoint that you transition to FW after? I can see why it makes sense, but it's called VTOL takeoff, we might be misinterpreting that.

**Comment by LorenzMeier** [comment]
Is this ready to go?

**Comment by sanderux** [comment]
No. I need to get in sync with @AndreasAntener  on how to approach this. As i have tried to implement / test this it is transitioning after takeoff and before navigating to the takeoff waypoint. I have integrated it in the vtol_land in a different check but that needs to move to it's own work item.

**Comment by sanderux** [comment]
@AndreasAntener 
>This adds a side effect to a check method. I would like to give this it's own work item that would be handled before the "need to move to land" check. So adding it before this line: https://github.com/PX4/Firmware/pull/3720/files#diff-9b5a77642a641b107c8f8a966de34acdR466

Link doesnt work, where do you want it?

**Comment by AndreasAntener** [comment]
@sanderux rebased and code moved, but completely untested. I also checked with Tridge quickly if was meant to transition right after vtol takeoff and it is, so your impl is right.

**Comment by LorenzMeier** [comment]
Can we rebase this, test this and get it in?

**Comment by sanderux** [comment]
No, not done yet. Currently not functional. I will get it done asap

**Comment by sanderux** [comment]
got replaced by https://github.com/PX4/Firmware/pull/4055

**Pull request closed without merging**